### PR TITLE
Issue/836 content negotiation

### PR DESCRIPTION
**Pull request by stefanorosanelli**

### CODE REVIEW

File: plugins/BEdita/API/config/bootstrap.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<?php
+/**
+ * BEdita - a semantic content management framework
```
Comment (batopa) [inline plugins/BEdita/API/config/bootstrap.php]: Wrong header

File: plugins/BEdita/API/src/View/JsonApiView.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * BEdita - a semantic content management framework
```
Comment (batopa) [inline plugins/BEdita/API/src/View/JsonApiView.php]: Wrong header

**Comment by batopa** [comment]
I have found an issue with request

``` http
GET /users
```

and no data on database.

The response is an internal server error **500**

``` json
{
    "error": {
        "status": "500",
        "title": "Key `id` is mandatory"
    }
}
```

I will expect a **200** with empty 'data'.

Here is a unit test to simulate it

``` php
public function testEmpty()
{
    $user = TableRegistry::get('Users');
    $user->deleteAll([]);
    $this->configRequest([
        'headers' => [
            'Accept' => 'application/vnd.api+json',
        ]
    ]);

    $this->get('/users');
    $result = json_decode($this->_response->body(), true);

    $this->assertResponseCode(200);
    $this->assertContentType('application/vnd.api+json');
    $this->assertEquals([], $result['data']);
}
```

**Comment by batopa** [comment]
Two things before merge the PR:
1. arranging pagination meta data as described in https://github.com/bedita/bedita/issues/839#issuecomment-217815166
2. as minor issue requests with header `Accept: text/html` (as requests by browser) shown a response containing
   
   ``` json
   {
   "users": [
       {
           "id": 4,
           "username": "hettinger.vida",
           "blocked": false,
           "last_login": null,
           "last_login_err": null,
           "num_login_err": 0,
           "created": "2016-05-09T13:37:30+0000",
           "modified": "2016-05-09T13:37:30+0000"
       },
       {
           "id": 5,
           "username": "sim67",
           "blocked": false,
           "last_login": null,
           "last_login_err": null,
           "num_login_err": 0,
           "created": "2016-05-09T13:37:30+0000",
           "modified": "2016-05-09T13:37:30+0000"
       }
   ]
   }
   ```
   
   Is it expected? I would expect the complete JSON API response.

**Comment by fquffio** [comment]
@batopa Both issues should have been addressed, now. I don't know why Travis CI build hasn't been triggered, but as soon as it will, you may proceed with tests.

HTML view has been achieved using a sub-request: all HTML requests to the API are now redirected to a single action (`\BEdita\API\Controller\AppController::html()`), and a sub-request is triggered with the current URL **but** with JSON Content-Type and Accept headers. This way, the very same serialization logic applies and therefore we are not required to duplicate it in the `Common/html_json.ctp` template anymore.

This approach should work for POST and authenticated requests as well, **but** I haven't configured the sub-request for such scenarios, yet, so we'll need to take care of it as soon as we actually need this feature.

**Pull request merged**